domain/application/state/state.go

```diff
@@ -277,3 +280,52 @@ VALUES ($M.unit_uuid, $M.net_node_uuid, $M.unit_id, $M.life_id, $M.application_u
 	}, nil
 
 }
+
+// StorageDefaults returns the default storage sources for a model.
+func (st *State) StorageDefaults(ctx context.Context) (domainstorage.StorageDefaults, error) {
+	rval := domainstorage.StorageDefaults{}
+
+	db, err := st.DB()
+	if err != nil {
+		return rval, errors.Trace(err)
+	}
+
+	attrs := []string{application.StorageDefaultBlockSourceKey, application.StorageDefaultFilesystemSourceKey}
+	attrsSlice := sqlair.S(transform.Slice(attrs, func(s string) any { return any(s) }))
+	stmt, err := sqlair.Prepare(`
+SELECT &KeyValue.* FROM model_config WHERE key IN ($S[:])
+`, sqlair.S{}, KeyValue{})
+	if err != nil {
+		return rval, errors.Trace(err)
+	}
+
+	return rval, db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		var values []KeyValue
+		err := tx.Query(ctx, stmt, attrsSlice).GetAll(&values)
+		if err != nil {
+			return fmt.Errorf("getting model config attrs for storage defaults: %w", err)
+		}
+
+		for _, kv := range values {
+			switch k := kv.Key; k {
+			case application.StorageDefaultBlockSourceKey:
+				v := fmt.Sprint(kv.Value)
+				rval.DefaultBlockSource = &v
+			case application.StorageDefaultFilesystemSourceKey:
+				v := fmt.Sprint(kv.Value)
+				rval.DefaultFilesystemSource = &v
+			}
+		}
+		return nil
+	})
+}
+
+// GetStoragePoolByName returns the storage pool with the specified name, returning an error
+// satisfying [storageerrors.PoolNotFoundError] if it doesn't exist.
+func (st *State) GetStoragePoolByName(ctx context.Context, name string) (domainstorage.StoragePoolDetails, error) {
+	db, err := st.DB()
+	if err != nil {
+		return domainstorage.StoragePoolDetails{}, errors.Trace(err)
+	}
+	return storagestate.GetStoragePoolByName(ctx, db, name)
```

Comment:
There are a lot of untraced errors through the storage pool loader infrastructure. We need to tidy that up.